openvdb_houdini/openvdb_houdini/SOP_OpenVDB_Merge.cc

```diff
@@ -0,0 +1,682 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+
+/// @file SOP_OpenVDB_Merge.cc
+///
+/// @authors Dan Bailey
+///
+/// @brief Merge OpenVDB grids.
+
+#include <houdini_utils/ParmFactory.h>
+#include <openvdb_houdini/Utils.h>
+#include <openvdb_houdini/PointUtils.h>
+#include <openvdb_houdini/SOP_NodeVDB.h>
+#include <openvdb_houdini/GEO_PrimVDB.h>
+
+#include <openvdb/points/PointDataGrid.h> // points::PointDataGrid
+#include <openvdb/tools/GridTransformer.h> // tools::replaceToMatch()
+#include <openvdb/tools/LevelSetRebuild.h> // tools::doLevelSetRebuild()
+#include <openvdb/tools/Merge.h>
+
+#include <UT/UT_Interrupt.h>
+#include <UT/UT_Version.h>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+
+using namespace openvdb;
+
+namespace hvdb = openvdb_houdini;
+namespace hutil = houdini_utils;
+
+
+class SOP_OpenVDB_Merge: public hvdb::SOP_NodeVDB
+{
+public:
+    SOP_OpenVDB_Merge(OP_Network*, const char* name, OP_Operator*);
+    ~SOP_OpenVDB_Merge() final {}
+
+    static OP_Node* factory(OP_Network*, const char* name, OP_Operator*);
+
+    class Cache: public SOP_VDBCacheOptions
+    {
+    public:
+        fpreal getTime() const { return mTime; }
+    protected:
+        OP_ERROR cookVDBSop(OP_Context&) final;
+    private:
+        fpreal mTime = 0.0;
+    };
+};
+
+
+void
+newSopOperator(OP_OperatorTable* table)
+{
+    if (table == nullptr) return;
+
+    hutil::ParmList parms;
+
+    parms.add(hutil::ParmFactory(PRM_STRING, "group", "Group")
+        .setTooltip("Specify a subset of the input VDBs to be modified.")
+        .setChoiceList(&hutil::PrimGroupMenuInput1)
+        .setDocumentation(
+            "A subset of the input VDBs to be modified"
+            " (see [specifying volumes|/model/volumes#group])"));
+
+    parms.add(hutil::ParmFactory(PRM_STRING, "collation", "Collation")
+        .setDefault("nameclassandtype")
+        .setChoiceListItems(PRM_CHOICELIST_SINGLE, {
+            "nameclassandtype", "Name, VDB Class and Value Type",
+            "classandtype",     "VDB Class and Value Type"
+        })
+        .setTooltip(
+            "Criteria under which groups of grids are merged.")
+        .setDocumentation(
+            "The criteria under which groups of grids are merged.\n\n"
+            "Grids with different names will be merged if name is not part"
+            " of the collation criteria"));
+
+    // Menu of resampling options
+    parms.add(hutil::ParmFactory(PRM_STRING, "resample", "Resample VDBs")
+        .setDefault("first")
+        .setChoiceListItems(PRM_CHOICELIST_SINGLE, {
+            "first",        "To Match First VDB"
+        })
+        .setTypeExtended(PRM_TYPE_JOIN_PAIR)
+        .setDocumentation("Specify which grid to use as reference when resampling."));
+
```

Comment:
Do we need to add this now if it doesn't have any options yet?  Isn't it safe just to not have it; and add it with First as the default when we have more than one choice?

Reply:
Fair point, I've made this an obsolete parameter for the time being.